Source/JavaScriptCore/runtime/FunctionConstructor.cpp

```diff
@@ -145,6 +145,11 @@ JSObject* constructFunction(JSGlobalObject* globalObject, const ArgList& args, c
     auto code = stringifyFunction(globalObject, args, functionName, functionConstructionMode, scope, functionConstructorParametersEndPosition);
     EXCEPTION_ASSERT(!!scope.exception() == code.isNull());
 
+    if (Options::useTrustedTypes() && globalObject->requiresTrustedTypes() && !globalObject->globalObjectMethodTable()->canCompileStrings(globalObject, CompilationType::Function, code, args, jsUndefined())) {
```

Comment:
It will only regress when TT is enforced through CSP. We probably can't cache either because it can end up executing JS callbacks which could be strateful (no guarantee it will return the same result each call).

Reply:
When is this `canCompileStrings` used? Is it super rare?
Function constructor is very hot code. And if it is not rare, you should make most of part inlined in JSC side and make it almost zero-cost. If this "requiresTrustedTypes" condition is super rare in the wild, then it is fine.